modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/LogicalRelImplementor.java

```diff
@@ -343,11 +342,11 @@ public Node<RowT> visit(IgniteIndexScan rel) {
         if (searchBounds != null) {
             Comparator<RowT> searchRowComparator = null;
 
-            if (idx.collations() != null) {
-                searchRowComparator = expressionFactory.comparator(TraitUtils.createCollation(idx.collations()));
+            if (idx.type() != IgniteIndex.Type.SORTED) {
```

Comment:
can`t understand this - we want hash index and build comparator according to DESC\ASC ordering ? I suppose that ordering is about sorted index, did i miss smth ?